### PR TITLE
Add certificates from Let's Encrypt

### PR DESCRIPTION
**Pull request by caalle**
I've started to update the proxy to see if it is possible to use Let's Encrypt. It looks like the [jrcs/letsencrypt-nginx-proxy-companion](https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion) docker image is one of the most popular right now.

@mskyttner @Inkimar @idali0226  what do you think? Have you done any more testing on this?

### CODE REVIEW

**Comment by mskyttner** [comment]
I used eforce21/letsencrypt-nginx-proxy (see hub.docker.com) in https://github.com/raquamaps/mirroreum/blob/master/docker-compose.yml successfully so far for a couple of weeks. It auto-updates certificates when they time out and has not gotten in the way. Numbers: it has 4.6K pulls and 5 stars. The one you look at has better numbers and is based on Alpine linux - so smaller and leaner, less surface and if it is equally simple to set up, it sounds like an improvement over the eforce21 variant. So I say go for it.

When it comes to nrm.se and dina-web.net we have wildcard certs already. Can we avoid those entirely? It would be nice, the less "secret files" the better and smoother deployments. Also no cost and admin for maintaining. Looking forward to hear how it works out!

**Comment by mskyttner** [comment]
I wasn't able to get the "jrcs companion" variant to work properly in production. I was struggling with some certificate errors and invalid certificates strangely enough. So I switched to the eforce variant which I found simpler to use, with less volume mappings and more "just works". It does not have as good numbers and it is a little bulkier, but it is based on the latest nginx-proxy stuff. With the changes above, I was able to get three different projects to automatically fetch and work with letsencrypt SSL certs (mirroreum, swedish bird trends and dw-chat). 

@caalle please see if this works also for dw-collections and merge it if it does?

File: README.md [hunk 1]
```diff
@@ -1,6 +1,38 @@
...
+
+Firstly, `docker` and `docker-compose` are required.
+
+Run only one instance of this component on the server to get a reverse proxy with SSL support in front of other services.
+
+The other services then only need to add three environment variables which will route traffic to them. In the `docker-compose.yml` file of the service, add something along these lines to expose it under a specific name with SSL support provided from `dw-proxy`:
+
+		environment:
+		  - VIRTUAL_HOST=beta-sso.dina-web.net
+		  - LETSENCRYPT_EMAIL=dina@mail.dina-web.net
```
Comment (caalle) [inline README.md:15]: Do I still need to set `- LETSENCRYPT_HOST=yourhost.something.com`?

**Comment by caalle** [comment]
Is this working locally for you? I'm getting `Failed authorization procedure. beta-api.dina-web.net (http-01): urn:acme:error:unauthorized :: The client lacks sufficient authorization :: Invalid response from http://beta-api.dina-web.net/.well-known/acme-challenge/L8F8Iuk-atB7JHWH5LS1kff6STUVb68ejDxrSNC2LJQ [178.62.213.220]: 404`

**Comment by idali0226** [comment]
I just tried with dw-collections, it is not working locally for me.

**Comment by mskyttner** [comment]
It might be tricky to get the cert generation to work locally without you having a tunnel set up where letsencrypt can make their callback? So running locally requires editing both /etc/hosts and punching a hole to the internet where letsencrypt can make their callback, two steps which are both non-trivial. 

But if you can find a guide to explain how to test letsencrypt cert generation locally, perhaps we can follow that?

Until then when running locally you might want to use existing certs? And for now limit testing of the cert generation to a server which has a public ip?

File: README.md (hunk 1 above)
Comment (mskyttner) [inline README.md:15]: Nope, that is for the other one. Also the LETSENCRYPT_EMAIL is optional and will default to info@VIRTUAL_HOST if not set, see https://hub.docker.com/r/eforce21/letsencrypt-nginx-proxy/

**Comment by idali0226** [comment]
It works on my local docker container.

**Comment by mskyttner** [comment]
I learned that you can make 5 requests using letsencrypt per week: https://community.letsencrypt.org/t/cannot-renew-certificate-too-many-requests/9190

.. by running into the limit, so now I cannot test cert generation for a while. 

I also ran into this (letsencrypt is happier with DNS A entries and doesn't like CNAME entries as much right now, it seems): https://community.letsencrypt.org/t/dns-problem-nxdomain-looking-up-a-for/16657

In any case, this seems to work. Should work also with own wildcard certs (put them in ./certs and .gitignore should have your back) and custom proxy wide header configs (put them in ./conf/dina.conf). 

@caalle please double check and merge at your convenience.

**Pull request closed without merging**